From 916a6fd3d2b05f4d706cde9d3f5f3ad9b71e6e25 Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Date: Wed, 4 Jan 2012 21:18:12 +0200
Subject: egl: test: Check that eglGetConfigs checks for NULL parameter

Passing NULL as the last paramter of eglGetConfigs is required to
generate EGL_BAD_PARAMETER. Add unit test to check that eglGetConfigs
functions correctly.

Category: AOSP Enhancement
Domain: GRAPHICS
Upstream-Candidate: yes

Change-Id: I22fb382ca84334237eafbe291e3f943d2045944d
Original-Change-Id: I9c998873c08fe60b2bcf6ae73da5954c79da3dda
Signed-off-by: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Signed-off-by: Russell Webb <russell.webb@intel.com>
---
 opengl/tests/EGLTest/EGL_test.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/opengl/tests/EGLTest/EGL_test.cpp b/opengl/tests/EGLTest/EGL_test.cpp
index f6644fb..c79fb5b 100644
--- a/opengl/tests/EGLTest/EGL_test.cpp
+++ b/opengl/tests/EGLTest/EGL_test.cpp
@@ -53,6 +53,13 @@ protected:
     }
 };
 
+TEST_F(EGLTest, EGLGetConfigsWithNullNumConfigs) {
+    EGLBoolean success;
+    success = eglGetConfigs(mEglDisplay, NULL, 0, NULL);
+    ASSERT_EQ(EGL_FALSE, success);
+    ASSERT_EQ(EGL_BAD_PARAMETER, eglGetError());
+}
+
 TEST_F(EGLTest, DISABLED_EGLConfigEightBitFirst) {
 
     EGLint numConfigs;
-- 
1.7.4.1

